src/os/ui/draw/draw.js

```diff
@@ -1,8 +1,166 @@
+
+  var cfg = {
+    lon: {
+      n: (extent[0] < extent[2]) ? extent[0] : extent[2],
+      x: (extent[0] < extent[2]) ? extent[2] : extent[0]
+    },
+    lat: {
+      n: (extent[1] < extent[3]) ? extent[1] : extent[3],
+      x: (extent[1] < extent[3]) ? extent[3] : extent[1]
+    }
+  }; // sort the data so the loops later are faster
+
+  var prop = {};
+  goog.object.extend(prop, feature.values_);
+  delete prop['geometry'];
+  delete prop['title'];
+  delete prop['_node'];
+
+  // TODO research a better, faster way to create/approximate the grid
+  // build detail x detail degree boxes that fit the "snapped" extent
+  for (var lon = cfg.lon.n; lon < cfg.lon.x; lon += detail) {
+    for (var lat = cfg.lat.n; lat < cfg.lat.x; lat += detail) {
+      var gridExtent = ol.proj.transformExtent(
+          [lon, lat, lon + detail, lat + detail],
+          os.proj.EPSG4326,
+          os.map.PROJECTION);
+
+      // only add this to the grid if the original polygon intersects it
+      if (feature.getGeometry().intersectsExtent(gridExtent)) {
+        features.push(os.ui.draw.gridFeatureFromExtent_(gridExtent, prop, options));
+      }
+    }
+  }
+
+  // TODO build a trace of the grid and save that to the Drawing Layers
+
+  return features;
+};
+
+
+/**
+ * Helper function; gets a numeric representation of the JSON setting
+ *
+ * @param {string} key
+ * @param {number} defaultValue
+ * @return {number}
+ */
+os.ui.draw.getGridSetting = function(key, defaultValue) {
+  var value = defaultValue;
+  try {
+    value = parseFloat(os.settings.get(key, defaultValue));
+  } catch (e) {
+    // do nothing
+  }
+  return value;
+};
+
+
+/**
+ * Build a grid feature from the extent and GridOptions
+ *
+ * @param {Array.<number>} extent The outer bounds of this grid
+ * @param {Object} prop The drawing and altitude modes; copied from the context feature
+ * @param {osx.ui.draw.GridOptions} options The color, line thickness, etc settings for this grid
+ * @return {ol.Feature}
+ * @private
+ */
+os.ui.draw.gridFeatureFromExtent_ = function(extent, prop, options) {
+  // new() is faster than doing os.feature.copyFeature(feature) from the original feature
+  var g = ol.geom.Polygon.fromExtent(extent);
+  var prop_ = {'geometry': g};
+  goog.object.extend(prop_, prop);
+
+  var f = new ol.Feature(prop_);
+  f.setId(goog.string.getRandomString());
+  f.setStyle(options.style);
+  f.set(os.data.RecordField.DRAWING_LAYER_NODE, false);
+  f.set(os.data.RecordField.INTERACTIVE, false);
+
+  return f;
+};
+
+
+/**
+ * Helper function to get valid default settings for grid capabilities
+ *
+ * @param {number=} detail The number of degrees squared used to draw the grid
+ * @param {number=} max The maximum number of grid squares to draw
+ * @param {ol.style.Style=} style The style applied to the squares of the grid
+ * @return {osx.ui.draw.GridOptions}
+ */
+os.ui.draw.gridOptionsInstance = function(detail, max, style) {
```

Comment:
I don't really like this here. I think it would be better to do the input checking in the function using it (`getGridFromFeature`). 

Since you only have 3 items that make up a `GridOptions` instance, those could just be parameters to `getGridFromFeature`. Not saying that's required, just that it would be simpler in this case.